go/src/koding/kites/kloud/stack/provider/builder.go

```diff
@@ -217,6 +217,20 @@ func (b *Builder) BuildStackTemplate(templateID string) error {
 	return nil
 }
 
+// Authorize verifies whether user is allowed to access b.StackTemplate.
+//
+// Prior calling to this method it is required to build the stack
+// template first.
+//
+// If the stack template was not built, the method is a nop.
+func (b *Builder) Authorize(username string) error {
+	if b.StackTemplate == nil {
```

Comment:
My point is that it is not feasible to add in a blanket manner nil-checks to every method that takes pointer argument - the nil-checks should be instead in places where we expect to have nil pointer. We do not expect to have nil argument in HasTemplateAccess, however we expect to have nil argument inside Authorize.